### ISSUE TITLE
Implement support for external Zofu

### ISSUE THREAD

**Issue by barracuda156** [josephalevin/fson#31]
CMakeLists have the following: https://github.com/josephalevin/fson/blob/16731d9b510b94a4e074a1fa7e35706f6045a85f/CMakeLists.txt#L133
In fact, however, `zofu` does have CMake implementation.
It is also desirable to add an option for using external `zofu`, so that it does not need to built twice. (Moreover, on macOS internal build of `zofu` just fails, at least on my system. External works perfectly and passes all tests.)

**Comment by acroucher** [comment]
Yes, `zofu` does have a CMake implementation now, so that could probably be changed.

@kprussing, you did the CMake stuff for FSON, would you be interesting in looking into this?

**Comment by barracuda156** [comment]
Besides, with `meson` everything worked fine with external `zofu` as is. Only CMake needs an option or a fix for it to work automatically.

**Comment by kprussing** [comment]
I'm not 100% sure what is going on with the built version of ZOFU. I just tried a handful of times on macOS and had no trouble building. I'm using gfortran 12.2.0 from MacPorts.
